### PR TITLE
feat: add support for generating `nice-grpc` server and client stubs

### LINKED ISSUE #545: Feature request: First class support for nice-grpc
Ts-proto currently has the "generic" service output target that creates a data representation of a service that can be used both at runtime, and to infer Typescript type signatures of services and clients.

With the magic of Typescript, this works with respect to type checking. An implementation of a service in nice-grpc that implements `ServiceImplementation<typeof SomeServiceDefinition>` will be appropriately typed and any mismatch in signature will be flagged.

It is not however that IDE friently. Specifically due to the inferred type signatures, there's no clean way to simply use "implement interface" or similar refactoring tools in IntelliJ or other IDEs. It is possible, but the signature it generates is far from readable.

To support nice-grpc, it would be extremely helpful if a regular Typescript interface could be generated, matching the signature of ServiceImplementation<typeof SomeServiceDefinition> and extending from it, so that an implementing class could simply provide the methods of this regular interface and typing errors would be much more readable.

### PR DESCRIPTION
**Pull request by aikoven**
This PR adds support for generating client and server interfaces for [nice-grpc](https://github.com/deeplay-io/nice-grpc). 

Currently `nice-grpc` generates these interfaces at compile time from generic definitions using TypeScript mapped types. This approach has drawbacks, mainly in developer experience, making type errors and editor hints less readable.

Closes #545

### CODE REVIEW

File: README.markdown
Comment (aikoven) [inline README.markdown:365]: Not sure about the latter, maybe it would be better to generate generic definitions implicitly on `outputServices=nice-grpc`.

File: src/main.ts
Comment (aikoven) [inline src/main.ts:320]: I needed to introduce a type that would be conditionally rendered once per file only when used. Could not find a better way than adding it to the global `Utils` object.

**Review by stephenh — APPROVED** [review]
Nice! Looks great; FWIW @aikoven anything that you want to ship in ts-proto to make it "more out of the box" for nice-grpc and/or just in general is great imo.

Also feel free to ping another maintainer for a +1 & merge without me, if I take too long to get to a PR.

**Comment by aikoven** [comment]
It seems to have conflicted with #552, the build is failing after merge. Will make a fix tomorrow.

**Pull request merged**
